src/pytorch_lightning/core/module.py

```diff
@@ -193,25 +209,25 @@ def example_input_array(self, example: Any) -> None:
     @property
     def current_epoch(self) -> int:
         """The current epoch in the ``Trainer``, or 0 if not attached."""
-        return self.trainer.current_epoch if self.trainer else 0
+        return self.trainer.current_epoch if self._trainer else 0
```

Comment:
not sure why this changed?

Reply:
> Disambiguates the trainer attribute (optional) from the property getter (non-optional)

Here we want to use the optional attribute, since `current_epoch` can be accessed without a Trainer. For example

```
my_model = MyModel()
print(my_model.current_epoch)
``